File: test_archives/py_tests/batch1/test_function0.py
from functions.function0 import function0
import pytest
def test_function0_1():
  assert function0(1, 2.3, 'hello', True) == 3.3

def test_function0_2():
  assert function0(2, 4.6, 'world', False) == 0

def test_function0_3():
  assert function0(3, 6.9, 'hello', False) == 6.9

def test_function0_4():
  assert function0(4, 9.2, 'world', True) == 4

File: test_archives/cpp_tests/batch1/test_function0.cpp
#include "../catch.hpp"
#include "../functions/function0.cpp"

TEST_CASE("Test function0_1", "[function0]") {
	REQUIRE(function0(1, 2.3, "hello", true) == 3.3);
}

TEST_CASE("Test function0_2", "[function0]") {
	REQUIRE(function0(2, 4.6, "world", false) == 0);
}

TEST_CASE("Test function0_3", "[function0]") {
	REQUIRE(function0(3, 6.9, "hello", false) == 6.9);
}

TEST_CASE("Test function0_4", "[function0]") {
	REQUIRE(function0(4, 9.2, "world", true) == 4);
}